client/state/posts/utils.js

```diff
@@ -241,6 +243,33 @@ export function normalizeTermsForApi( post ) {
 }
 
 /**
+ * Returns truthy if local terms object is different than API response
+ *
+ * @param  {Object}  localTermEdits local state of term edits
+ * @param  {Object}  savedTerms     term object returned from API POST
+ * @return {Boolean}                are there differences in local edits vs saved terms
+ */
+export function hasTermEditDifferences( localTermEdits, savedTerms ) {
+	if ( ! localTermEdits || ! savedTerms ) {
+		return false;
+	}
+
+	return !! find( localTermEdits, ( terms, taxonomy ) => {
+		const termsArray = toArray( terms );
+		const isTag = ! isPlainObject( termsArray[ 0 ] );
+		const normalizedEditedTerms = isTag ? termsArray.sort() : map( termsArray, 'ID' ).sort();
+		const normalizedKey = isTag ? 'name' : 'ID';
+		const normalizedSavedTerms = map( savedTerms[ taxonomy ], normalizedKey ).sort();
```

Comment:
To avoid the need to compare terms in perfect sorted order, [`_.xor`](https://lodash.com/docs/4.17.4#xor) might come in handy:

```js
export function isTermsEqual( localTermEdits, savedTerms ) {
	return every( localTermEdits, ( terms, taxonomy ) => {
		const termsArray = toArray( terms );
		const isHierarchical = isPlainObject( termsArray[ 0 ] );
		const normalizedEditedTerms = isHierarchical ? map( termsArray, 'ID' ) : termsArray;
		const normalizedKey = isHierarchical ? 'ID' : 'name';
		const normalizedSavedTerms = map( savedTerms[ taxonomy ], normalizedKey );
		return ! xor( normalizedEditedTerms, normalizedSavedTerms ).length;
	} );
}
```